test/main_dummy_controller_test.cpp

```diff
@@ -0,0 +1,46 @@
+#include "ostream"
+#include <dummy_controller_c.hpp>
+
+#define CATCH_CONFIG_MAIN
+#include <catch.hpp>
+
+TEST_CASE("testing buttons", " [interface]") {
+    auto dummy_controller = r2d2::manual_control::dummy_controller_c();
+    for (int x = r2d2::manual_control::button_a;
+         x <= r2d2::manual_control::menu_right; x++) {
+        // check buttons
+        r2d2::manual_control::buttons button =
+            static_cast<r2d2::manual_control::buttons>(x);
+        CHECK(dummy_controller.get_button(button) == 0);
```

Comment:
Can you change the `Check()` to `REQUIRE()` as this will fail the test when the require fails instead of continuing.

Reply:
I read up on the difference between check and require and it said that check is used when its useful to know the result of all the tests, even if for example the first test fails. In this case i think it's useful to know the result of each test so you know which buttons, sliders and joysticks do, and do not work.